jiant/modules/seq2seq_decoder.py

```diff
@@ -148,9 +208,15 @@ def forward(
         """
         batch_size, _, _ = encoder_outputs.size()
 
-        use_gold = target_tokens is not None and self.training
+        if self.training:
+            assert target_tokens
+            mode = "train"
+        elif target_tokens is not None:
+            mode = "validation"
+        else:
+            mode = "test"
 
-        if use_gold:
+        if mode == "train" or mode == "validation":
             targets = target_tokens["words"]
```

Comment:
I would prefer to just remove this from the `if` condition, and then do something like `target_sequence_length = targets.size(1) - 1 if mode in ["train", "validation"] else self._max_decoding_steps`.

My mild concern is the presence of lots of code blocks that aren't totally obvious what they each do. This would get rid of a block.

Reply:
Done.